### ISSUE TITLE
Error: rpc error: code = Canceled desc = context canceled

### ISSUE THREAD

**Issue by KalanaDananjaya** [terraform-providers/terraform-provider-azurerm#12112]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
v0.14.6

### Debug Output

https://gist.github.com/KalanaDananjaya/a552eecb0a072eb116cea0e8f32e47c1

--->

### Expected Behaviour

TF files should run to the end and deploy the resources succesfully

### Actual Behaviour

RPC transport is breaking while applying changes

I am using following provider versions
- hashicorp/azuread "1.2.2"
- hashicorp/random  "2.2.0"
- microsoft/azuredevops  "0.1.0"
- hashicorp/azurerm v "2.49.0"

**Comment by magodo** [comment]
@KalanaDananjaya Could you please provide more context (e.g. steps to reproduce) for this issue? From your log provided, it looks like the user has canceled the execution of Terraform (e.g. by pressing <kbd>Ctrl-C</kbd>)

**Comment by KalanaDananjaya** [comment]
@magodo  thanks for checking this. I understand that it would be hard to debug without all the details, but I am unable to post my code here as it is from a private repo. I will try to give you as much context as possible and post any non-private code details to help that.

- Applying of TF changes for this private repo is working fine for the older users
- I tried to follow the same steps and my RPC transport gets broken in the middle as shown above. I have not cancelled the execution of terraform and I have tried this a lot of times with the same results. The execution gets cancelled automatically
- One of the older users tried deleting his .terraform folder and reapply the changes and now he is also getting the same issue

As this is working for the people with old .terraform directory, my guess is that this is something related to a provider but I can't figure out which one.
